### DISCUSSION TITLE
Select screen cursor *jumps* in Ikemen, it *scrolls* instead in Mugen 1.1

### DISCUSSION THREAD

**Discussion by Kasasagi77** [ikemen-engine/Ikemen-GO#1377] [Feature Request] [answered]
### Describe the issue

In the select screen, the cursor **scrolls smoothly** in the original Mugen 1.1 engine, in Ikemen it simply **jumps** instead.

### To Reproduce

1. Enter select screen of almost any screenpack.
2. Observe thew cursor behaviour when selecting different characters.
3. Compare visual effects between Mugen 1.1 and Ikemen.

### Expected behavior

The select screen cursor smoothly scrolls (possibly only when moving to an adjacent cell).

### Engine Version (or source code date)

2023-08-23

### Operating system

Linux

**Comment by MUGEN-Slayer** [comment]
It works like MUGEN 1.0. Taking into account when you go from the first slot of a row to the last and the cursor runs all over the screen, the current behavior is better.

**Comment by Kasasagi77** [comment]
Well, I think think that 'better' is at least relative in this case, but I can see that Mugen 1.0 indeed does not scroll, while Mugen 1.1 does.

I think that scrolling would at least deserve an option.

Or maybe Ikemen could simply combine the best of both worlds betweem Mugen 1.0 and 1.1 and scroll only when moving the cursor to an adjacent cell.

**Comment by Kasasagi77** [comment]
I have updated the description to reflect the above discussion.

**Comment by MUGEN-Slayer** [comment]
When I said "better", I should be more specific. In the context of MUGEN it's better because some screenpacks use a workaround to create the illusion of diagonal navigation.
![NESR002](https://github.com/ikemen-engine/Ikemen-GO/assets/68881657/0e0ecfdf-8024-4dda-9481-393dfcd05555)

This illusion is broken in 1.1. But I digress, as due to new features IKEMEN wouldn't have to worry. My bad.
That said, the current behavior isn't a bug per se, so if you'd like this to be a feature, I suppose "discussion" would be the most appropriate place since that's the area for "feature requests".

**Comment by Kasasagi77** [comment]
I at least do not agree that it is a pure "feature request", as this issue documents a clear, version-dependent, difference of behaviour between Ikemen and Mugen.

I think that someone with permissions to assign or change any labels, could simply classify this issue as something other than a pure "bug".

**Comment by Kasasagi77** [comment]
Hello @K4thos.

From my understanding this is a very direct report of a difference in behaviour between Ikemen and Mugen 1.1, in the scope of screenpack effects.

Can I ask why it was demoted from an "Issue" to a "Discussion"?

I understand that fixing this in a crude or simplistic way can potencially break compatibility with some very specific existing screenpacks, if the above comments are not taken into account, however the current `develop` for example breaks compatibility with basically all existing Ikemen stages in the name of Mugen compatibility and such breaking of compatibility is not questioned.

**Comment by K4thos** [comment, marked as answer]
implement by #2604
